server.py

```diff
@@ -50,6 +50,7 @@ def speaker(speaker):
     speaker=speaker.split("-")
     speaker=" ".join(speaker)
     speakers=Speaker.query.filter(Speaker.speaker_name == speaker).first()
+    # Why not use Speaker.talks?
     talks=Talk.query.filter(Talk.speakerID==speakers.speaker_id).all()

```

Comment:
You have the `Speaker.talks` relationship on your models. You can simplify this code by using it.